R/cast.R

```diff
@@ -133,6 +133,15 @@ vec_cast.default <- function(x, to) {
   stop_incompatible_cast(x, to)
 }
 
+# Cast `x` to `to` but only if they are coercible
+vec_coercible_cast <- function(x, to, ..., x_arg = "", to_arg = "") {
+  if (!missing(...)) {
+    ellipsis::check_dots_empty()
+  }
+  vec_type2(to, x, x_arg = to_arg, y_arg = x_arg)
```

Comment:
This is being called purely for its side effect of generating an error if there's no common type?

Reply:
Correct, I'll add a comment.